### ISSUE TITLE
Does not work on Fedora 28+

### ISSUE THREAD

**Issue by tuaris** [danstiner/rust-u2f#31]
I was unable to get this working on Fedora 28 and 29 using the instructions provided in the readme.  Installation completed successfully but no U2F devices are dected by browsers or yubi key utilities.

**Comment by tuaris** [comment]
User daemon is running

```
systemctl --user status softu2f
● softu2f.service - Software-only U2F Emulation Service
   Loaded: loaded (/usr/lib/systemd/user/softu2f.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2019-01-29 22:36:39 EST; 25s ago
 Main PID: 1677 (user-daemon)
   CGroup: /user.slice/user-1000.slice/user@1000.service/softu2f.service
           └─1677 /usr/libexec/softu2f/user-daemon

Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.126 INFO Started SoftU2f Session, store_path: /home/dmorante/.softu2f-secrets.json
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.126 INFO Opening socket
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.129 INFO Sending create device request, request: CreateDeviceRequest
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.136 INFO Sent create device request
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.136 INFO Got response, response: CreateDeviceResponse(Success)
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.137 TRCE Poll U2FHID
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.137 TRCE Ensure sink is ready
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.137 TRCE Try to step state machine
Jan 29 22:36:39 acquia-server2.morante.com user-daemon[1677]: Jan 29 22:36:39.137 TRCE Poll read from transport stream
Jan 29 22:36:39 acquia-server2.morante.com systemd[1666]: Started Software-only U2F Emulation Service.
```
System daemon is okay also

```
systemctl status softu2f
● softu2f.service - Software-only U2F Emulation Service
   Loaded: loaded (/usr/lib/systemd/system/softu2f.service; static; vendor preset: enabled)
   Active: active (running) since Tue 2019-01-29 22:36:26 EST; 1min 16s ago
 Main PID: 1156 (system-daemon)
    Tasks: 2 (limit: 2325)
   Memory: 1.8M
   CGroup: /system.slice/softu2f.service
           └─1156 /usr/libexec/softu2f/system-daemon

Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.131 DEBG state initialized, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.131 INFO run, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.131 DEBG poll running, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.131 TRCE poll => WouldBlock, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.132 TRCE poll_complete, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.134 DEBG poll, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.134 DEBG running, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.134 TRCE poll => Ok, frame: Start, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.134 TRCE poll => WouldBlock, uid: 1000
Jan 29 22:36:39 acquia-server2.morante.com system-daemon[1156]: Jan 29 22:36:39.134 TRCE poll_complete, uid: 1000
```

**Comment by tuaris** [comment]
Couldn't get it to work on 27 either.  There must be something I'm missing.

**Comment by danstiner** [comment]
Interesting. I think the Yubico demo site has changed and no longer works with my code, possibly because the demo site seems to be based on the Webauthn spec now instead of U2F. However I just tried and things still work on GitHub and other sites, so if you want to play around I would recommend that for now. I may try to dig deeper if I have time.

Caveat: This code is in no way "production-ready" and does not have the same security as a hardware key, it is a proof of concept I did for fun.
